examples/ccvm_boxqp_plot_ETS.py

```diff
@@ -0,0 +1,64 @@
+import glob
+import os
+from ccvm_simulators.ccvmplotlib import ccvmplotlib
+from ccvm_simulators.problem_classes.boxqp import ProblemInstance
+from ccvm_simulators.metadata import Metadata
+from ccvm_simulators.solvers import DLSolver
+
+import matplotlib.pyplot as plt
+
+METADATA_DIR = "./metadata"
+TEST_OUTPUT_DEST = f"{METADATA_DIR}/DL-CCVM_LGFGS_cpu_test.txt"
+TEST_INSTANCES_DIR = "./examples/test_instances/"
+PLOT_OUTPUT_DIR = "./plots"
+PLOT_OUTPUT_DEST = f"{PLOT_OUTPUT_DIR}/DL-CCVM_TTS_cpu_plot.png"
+
+
+if __name__ == "__main__":
+    # Initialize solver
+    batch_size = 1000
+    solver = DLSolver(device="cpu", batch_size=batch_size)  # or "cuda"
+
+    # Supply solver parameters for different problem sizes
+    solver.parameter_key = {
+        10: {"pump": 1.0, "dt": 0.001, "iterations": 10000, "noise_ratio": 15},
+        20: {"pump": 2.0, "dt": 0.005, "iterations": 15000, "noise_ratio": 10},
+    }
+
+    metadata_obj = Metadata(device=solver.device)
+    # Load test instances to solve
+    test_instances_files = [f for f in glob.glob(TEST_INSTANCES_DIR + "*.in")]
+    for instance_file in test_instances_files:
+        # Load the problem from the problem file into the instance
+        boxqp_instance = ProblemInstance(
+            instance_type="test",
+            file_path=instance_file,
+            device=solver.device,
+        )
+
+        # Scale the problem's coefficients for more stable optimization
+        boxqp_instance.scale_coefs(solver.get_scaling_factor(boxqp_instance.q_matrix))
+
+        # Solve the problem
+        solution = solver(
+            instance=boxqp_instance,
+            post_processor=None,
+        )
+        # Add metadata to list
+        metadata_obj.add_to_result_metadata(solution.get_metadata_dict())
+
+    # Save metadata to file
+    metadata_filepath = metadata_obj.save_metadata_to_file(METADATA_DIR)
+
+    
+    # Customize machine_parameters
+    machine_parameters = {
+        "gpu_power": {20: 28.43, 30: 29.8, 40: 31.09, 50: 31.29, 60: 31.49, 70: 32.22}
+    }
+    # TODO: This will be done in issue #137. Currently, just for testing purposes
+    ccvmplotlib.plot_ETS(
+        metadata_filepath=metadata_filepath,
+        problem="BoxQP",
+        energy_max_func=solver.cuda_energy_max(mechine_parameters=machine_parameters),
```

Comment:
Right now the user needs to know that right name for the callable function `cuda_energy_max` or `optics_energy_max` and also provide the right parameters for `machine_parameters`. 
I'm wondering if we can have a function in `solver` with a more general name such as `energy_max` where it receives a variable `machine` as input, in addition to `machine_parameters`, as chooses the appropriate callable based on the value of `machine`. 
Maybe discuss this with @KatieO4t1qbit . It's okay If you think this is not doable or more confusing.

Reply:
Oh, Farhad I like that 👍 
Estella, let me know if you want to chat about it.
Also, @ChanWooY-1QBit PR #143 might need a small update to the example file with this change.

Reply:
So this line would become 
```suggestion
        energy_max_func=solver.energy_max(machine="cuda", machine_parameters=machine_parameters),
```

Reply:
Another note, @Estella-Wang-1qbit can we please do these implementations within the solvers in another PR?
I should have made this ticket sooner to avoid confusion, but I've made it now: https://github.com/1QB-Information-Technologies/ccvm/issues/157

Reply:
Certainly! I'll include these changes in a separate PR. Thanks for making that ticket.

Reply:
Solver related code have been transferred to this new [PR.](https://github.com/1QB-Information-Technologies/ccvm/pull/168)